crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -47,164 +51,201 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
         let comments = f.context().comments().clone();
         let source = f.context().source();
 
-        let mut iter = statements.iter();
-        let Some(first) = iter.next() else {
-            return Ok(());
-        };
-
         let mut f = WithNodeLevel::new(node_level, f);
+        write!(
+            f,
+            [format_with(|f| {
+                let mut iter = statements.iter();
+                let Some(first) = iter.next() else {
+                    return Ok(());
+                };
 
-        // Format the first statement in the body, which often has special formatting rules.
-        let mut last = first;
-        match self.kind {
-            SuiteKind::Other => {
-                if is_class_or_function_definition(first) && !comments.has_leading_comments(first) {
-                    // Add an empty line for any nested functions or classes defined within
-                    // non-function or class compound statements, e.g., this is stable formatting:
-                    // ```python
-                    // if True:
-                    //
-                    //     def test():
-                    //         ...
-                    // ```
-                    write!(f, [empty_line()])?;
-                }
-                write!(f, [first.format()])?;
```

Comment:
I had to move the `first.format` out to avoid having to handle suppression comments in each branch.